viennacl/backend/mem_handle.hpp

```diff
@@ -86,6 +86,7 @@ inline memory_types default_memory_type(memory_types new_memory_type) { return d
  * Instead, this class collects all the necessary conditional compilations.
  *
  */
+template <typename OCLHandle>
```

Comment:
* Suggest renaming `OCLHandle` -> `HandleImpl`?

Maybe
```c++
template <typename OCLHandle =
#if defined(VIENNACL_WITH_OPENCL)
   stuff
#elif defined(VIENNACL_WITH_CUDA)
   stuff
#endif
>
class mem_handle_with_impl
{ ... };

typedef mem_handle_with_impl<> mem_handle;
```

This might also avoid having to add many of the angle brackets?